modules/sdk-core/src/bitgo/enterprise/enterprises.ts

```diff
@@ -54,4 +56,41 @@ export class Enterprises implements IEnterprises {
     const enterpriseData = (await this.bitgo.post(this.bitgo.url(`/enterprise`)).send(params).result()) as any;
     return new Enterprise(this.bitgo, this.baseCoin, enterpriseData);
   }
+
+  /**
+   * Resign TSS configs of all enterprises belonging to the user with a
+   * new user password / ecdh keychain
+   */
+  public async resignTssConfigsForEnterprises(
+    oldEcdhKeypair: EcdhDerivedKeypair,
+    newEcdhKeypair: EcdhDerivedKeypair
+  ): Promise<void> {
+    const enterpriseTssConfigsForUser = await this.bitgo
+      .get(this.bitgo.url('/tssconfigs/ecdsa', 2))
+      .query({
+        createdBy: 'me',
+      })
+      .send()
+      .result();
+
+    for (const enterpriseTssConfig of enterpriseTssConfigsForUser) {
+      const enterprise = new Enterprise(this.bitgo, this.bitgo.coin('tbtc'), {
+        id: enterpriseTssConfig.enterpriseId,
+        name: '',
+      });
+      const enterpriseChallenge = enterpriseTssConfig.ecdsa.challenge
+        ?.enterprise as EcdsaTypes.SerializedNtildeWithVerifiers;
+      const bitgoNitroChallenge = enterpriseTssConfig.ecdsa.challenge
+        ?.bitgoNitroHsm as EcdsaTypes.SerializedNtildeWithVerifiers;
+      const bitgoInstChallenge = enterpriseTssConfig.ecdsa.challenge
+        ?.bitgoInstitutionalHsm as EcdsaTypes.SerializedNtildeWithVerifiers;
+      await enterprise.resignEnterpriseChallenges(
+        oldEcdhKeypair,
+        newEcdhKeypair,
+        enterpriseChallenge,
+        bitgoInstChallenge,
+        bitgoNitroChallenge
+      );
+    }
```

Comment:
Put the promises in a `promise.all`. Is there a reason you need to do these all synchronously?